source/server/admin/prometheus_stats.cc

```diff
@@ -274,6 +276,35 @@ uint64_t outputPrimitiveStatType(Buffer::Instance& response, const StatsParams&
   return result;
 }
 
+/*
+ * Returns the prometheus output for a summary. The output is a multi-line string (with embedded
+ * newlines) that contains all the individual quantile values and sum/count for a single histogram
+ * (metric_name plus all tags).
+ */
+std::string generateSummaryOutput(const Stats::ParentHistogram& histogram,
+                                  const std::string& prefixed_tag_extracted_name) {
+  const std::string tags = PrometheusStatsFormatter::formattedTags(histogram.tags());
+  const std::string hist_tags = histogram.tags().empty() ? EMPTY_STRING : (tags + ",");
+
+  const Stats::HistogramStatistics& stats = histogram.cumulativeStatistics();
+  Stats::ConstSupportedBuckets& supported_quantiles = stats.supportedQuantiles();
+  const std::vector<double>& computed_quantiles = stats.computedQuantiles();
+  std::string output;
+  for (size_t i = 0; i < supported_quantiles.size(); ++i) {
+    double quantile = supported_quantiles[i];
+    double value = computed_quantiles[i];
+    output.append(fmt::format("{0}{{{1}quantile=\"{2}\"}} {3}\n", prefixed_tag_extracted_name,
```

Comment:
I'm not sure what the default formatting for a double is for `{3}` here. Do you think some specific precision should be specified?

Reply:
Yeah, that's probably a good idea, although I'm a little confused by [this comment](https://github.com/envoyproxy/envoy/blob/3164f06df3269e5041f1a6f0e650bfbe94f25653/source/server/admin/prometheus_stats.cc#L112-L116) about fixed-point not being supported, since it seems pretty well documented in the [fmt library](https://hackingcpp.com/cpp/libs/fmt.html#fp-numbers)? Not sure what a sane default precision should be... first thought was to use `.32g` like in the bucket output above.

Reply:
Honestly, not sure about the fmtlib `f` format vs that comment. I'm pretty sure I wrote the envoy comment. But I don't recall the situation. Probably either that I completely missed `f` in the docs, or it didn't work in the way we needed (maybe not enough significant digits or something?). 

I'd say for now use the existing `.32g` format, and someone should circle back in a separate change and see if we can switch them all to something better or not.